File: backend/java/src/main/java/de/iits/elocalculatorbackend/match/model/dto/MatchRequestDto.java
package de.iits.elocalculatorbackend.match.model.dto;

import de.iits.elocalculatorbackend.match.model.entity.Match;
import de.iits.elocalculatorbackend.match.model.enumeration.Outcome;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;

public record MatchRequestDto (
    @NotNull String whitePlayerUsername,
    @NotNull String blackPlayerUsername,
    @NotNull Outcome outcome
) {

    public Match toEntity() {
        var match = new Match();
        match.setBlackPlayerUsername(blackPlayerUsername);
        match.setWhitePlayerUsername(whitePlayerUsername);
        match.setOutcome(outcome);
        match.setPlayedOn(LocalDateTime.now());
        return match;
    }
}

File: backend/springboot-kotlin/src/main/kotlin/de/iits/elo/match/model/dto/MatchRequestDto.kt
package de.iits.elo.match.model.dto

import de.iits.elo.match.model.entity.Match
import de.iits.elo.match.model.enumeration.Outcome

data class MatchRequestDto(

    val whitePlayerUsername: String,
    val blackPlayerUsername: String,
    val outcome: Outcome,
)

fun MatchRequestDto.toEntity() =
    Match(
        whitePlayerUsername = whitePlayerUsername,
        blackPlayerUsername = blackPlayerUsername,
        outcome = outcome,
    )
